be/src/exec/vectorized/aggregate/distinct_blocking_node.cpp

```diff
@@ -136,39 +138,57 @@ std::vector<std::shared_ptr<pipeline::OperatorFactory> > DistinctBlockingNode::d
 
     OpFactories ops_with_sink = _children[0]->decompose_to_pipeline(context);
 
-    // Create a shared RefCountedRuntimeFilterCollector
-    auto&& rc_rf_probe_collector = std::make_shared<RcRfProbeCollector>(2, std::move(this->runtime_filter_collector()));
-
     // shared by sink operator and source operator
-    AggregatorFactoryPtr aggregator_factory = std::make_shared<AggregatorFactory>(_tnode);
 
-    auto partition_expr_ctxs = std::move(_group_by_expr_ctxs);
-    auto sink_operator = std::make_shared<AggregateDistinctBlockingSinkOperatorFactory>(
-            context->next_operator_id(), id(), aggregator_factory, partition_expr_ctxs);
+    auto should_cache = context->should_interpolate_cache_operator(ops_with_sink[0], id());
+    auto operators_generator = [this, should_cache, &context](bool post_cache) {
+        AggregatorFactoryPtr aggregator_factory = std::make_shared<AggregatorFactory>(_tnode);
+        AggrMode aggr_mode = should_cache ? (post_cache ? AM_BLOCKING_POST_CACHE : AM_BLOCKING_PRE_CACHE) : AM_DEFAULT;
+        aggregator_factory->set_aggr_mode(aggr_mode);
+        std::vector<ExprContext*> partition_expr_ctxs;
+        Expr::create_expr_trees(_pool, _tnode.agg_node.grouping_exprs, &partition_expr_ctxs);
```

Comment:
we should check return status

Reply:
Frankly speaking, return status should be checked generally, but here, group_by_exprs are just SlotRef exprs.  I think it is sound and safe. by the way, decompose_to_pipelines functions do not have a error-handling logic, later, another pr should be submitted to resolve this problem, it is not intention of this pr.